[pickers] Remove the `defaultCalendarMonth` prop (#10987)
mui/mui-x: docs/data/migration/migration-pickers-v6/migration-pickers-v6.md

The same applies to `slotProps` and `componentsProps`.
:::

### Replace `defaultCalendarMonth` with `referenceDate`

The `defaultCalendarMonth` has been removed in favor of the more flexible `referenceDate` prop.

> It was available in `DateCalendar` and all the pickers using it for date editing.

Reviewer: ```suggestion
```

Is it necessary? People already using it are not concerned by this change

Author: My goal here was to help people know if they can skip this section

Author: Removed

> The new `referenceDate` prop is not limited to the default month, learn more on this prop on [the `DateCalendar` doc](/x/react-date-pickers/date-calendar/#choose-the-initial-year-month) or [the `referenceDate` doc](/x/react-date-pickers/base-concepts/#reference-date-when-no-value-is-defined).

Reviewer: ```suggestion
:::warning
The new `referenceDate` prop is not limited to the default month.
It will also impact year, day, and time.

Learn more on this prop on [the `DateCalendar` doc](/x/react-date-pickers/date-calendar/#choose-the-initial-year-month) or [the `referenceDate` doc](/x/react-date-pickers/base-concepts/#reference-date-when-no-value-is-defined).
:::
```

Not sure about the warning. But I would recomand to first point explicitly what could be modified by this change

+ } from '@mui/x-date-pickers/PickersCalendarHeader';
```

## Removed props

### Replace `defaultCalendarMonth` with `referenceDate`

> The `defaultCalendarMonth` has been removed in favor of the more flexible `referenceDate` prop.

Reviewer 2: ```suggestion
The deprecated `defaultCalendarMonth` prop has been removed in favor of the more flexible `referenceDate` prop.
```

> :::warning

Reviewer 2: WDYT about going with `info` callout instead? 🤔 
```suggestion
:::info
```

The new `referenceDate` prop is not limited to the default month.
It will also impact year, day, and time.

> Learn more on this prop on [the `DateCalendar` doc](/x/react-date-pickers/date-calendar/#choose-the-initial-year-month) or [the `referenceDate` doc](/x/react-date-pickers/base-concepts/#reference-date-when-no-value-is-defined).

Reviewer 2: ```suggestion
Learn more on this prop on [the `DateCalendar` documentation](/x/react-date-pickers/date-calendar/#choose-the-initial-year-month) or [the `referenceDate` documentation](/x/react-date-pickers/base-concepts/#reference-date-when-no-value-is-defined) pages.
```
